pkg/cli/cluster_free_disk_space.go

```diff
@@ -0,0 +1,223 @@
+		return fmt.Errorf("failed to get rook free space: %w", err)
+	}
+
+	requestedString := bytefmt.ByteSize(uint64(requested))
+	freeString := bytefmt.ByteSize(uint64(free))
+	if free < requested {
+		return fmt.Errorf("not enough space on rook (requested %s, available %s)", requestedString, freeString)
+	}
+	fmt.Printf("Available disk space found in rook: %s\n", freeString)
+	return nil
+}
+
+// NewClusterCheckFreeDiskSpaceCmd returns a command that is capable of reporting back the amount of free space in the cluster for a provided storage class.
+func NewClusterCheckFreeDiskSpaceCmd(cli CLI) *cobra.Command {
+	var forStorageClass, openEBSImage, openEBSNode, biggerThanString string
+	var biggerThanBytes int64
+	var clientSet kubernetes.Interface
+	var rookClientSet rookcli.Interface
+	var selectedClass *storagev1.StorageClass
+
+	cmd := &cobra.Command{
+		Use:          "check-free-disk-space",
+		Short:        "List and analyse the available disk space for a given Storage Class.",
+		SilenceUsage: true,
+		Example: fmt.Sprintf(""+
+			"In the following examples 'openebs' is the name of a storage class backed by the %s storage provisioner while 'rook' is the name of a storage\n"+
+			"class backed by the %s or %s storage provisioners.\n\n"+
+			"# reports the available space in rook\n"+
+			"kurl cluster check-free-disk-space --storageclass rook\n\n"+
+			"# reports the available space in all nodes\n"+
+			"kurl cluster check-free-disk-space --storageclass openebs\n\n"+
+			"# checks if there is 10G available on node node0\n"+
+			"kurl cluster check-free-disk-space --storageclass openebs --openebs-node-name node0 --openebs-image ubuntu:latest --bigger-than 10G\n\n"+
+			"# checks if there is 10G available on all nodes in the cluster\n"+
+			"kurl cluster check-free-disk-space --storageclass openebs --bigger-than 10G\n"+
+			"# checks if there is 20G available in the cluster on the default storage class\n"+
+			"kurl cluster check-free-disk-space --bigger-than 20G\n",
+			openEBSLocalProvisioner, rookRBDProvisioner, rookCephFSProvisioner,
+		),
+		Long: fmt.Sprintf(""+
+			"This program returns the amount of free disk space (in bytes) for a given storage class or compares if there is enough space to hold a certain\n"+
+			"amount of data (--bigger-than). For OpenEBS, when no --bigger-than flag is provided, this program returns a list of nodes and their respective\n"+
+			"free space, while for Rook storage only the available space is returned. When --bigger-than flag is used this program sets the exit code to zero\n"+
+			"if the space available is bigger than the quantity provided. For OpenEBS, if no node has been provided through --openebs-node-name the exit code\n"+
+			"will be zero only if all nodes free space are bigger than the provided quantity (see Examples section for more details).\n\n"+
+			"Supports the following storage provisioners: %s, %s, %s",
+			openEBSLocalProvisioner, rookRBDProvisioner, rookCephFSProvisioner,
```

Comment:
Same of above can we use a const for the text here?

Reply:
Same answer.